src/bridges/uniswapv3/AsyncUniswapV3Bridge.sol

```diff
@@ -0,0 +1,330 @@
+
+            //(,,,,,,,,,,uint128 tokensOwed0,uint128 tokensOwed1) = nonfungiblePositionManager.positions(deposits[_interactionNonce].tokenId);
+            //implement a fee percentage rather than approving whole thing?
+            // TODO: isn't using msg.sender a security vulnerability here given that the method doesn't do any entry checks and is called externally?
+            IERC20(deposits[_interactionNonce].token0).safeIncreaseAllowance(
+                msg.sender,
+                amount0 * (deposits[_interactionNonce].fee / 1000000)
+            );
+            // TODO: why is amount 0 here and not amount 1?
+            IERC20(deposits[_interactionNonce].token1).safeIncreaseAllowance(
+                msg.sender,
+                amount1 * (deposits[_interactionNonce].fee / 1000000)
+            );
+        }
+    }
+
+    /**
+     * @notice  Function used to cancel if an order is past its expiry
+     * @dev if an order is past its expiry, this fxn withdraws the liquidity
+     * @param _interactionNonce the nonce of the interaction
+     * @return amount0 the amount of token0 returned by withdrawal of liquidity
+     * @return amount1 the amount of token1 returned by withdrawal of liquidity
+     */
+
+    function cancel(uint256 _interactionNonce) external returns (uint256 amount0, uint256 amount1) {
+        if (asyncOrders[_interactionNonce].expiry >= block.timestamp) {
+            asyncOrders[_interactionNonce].finalisable = true;
+            (amount0, amount1) = _withdraw(_interactionNonce, deposits[_interactionNonce].liquidity);
+            //console.log(amount0, amount1, "amounts");
+            //user may now call processAsyncDefiInteraction successfully
+        }
+    }
+
+    /**
+     * @notice  convert function to that takes inputs and mints liquidity
+     * @dev takes an auxdata w/ ticklower, tickupper, and fee tier of pool.
+     * @param _inputAssetA input token
+     * @param _inputAssetB not used
+     * @param _outputAssetA output token
+     * @param _outputAssetB not used
+     * @param _inputValue size of input token
+     * @param _interactionNonce nonce
+     * @param _auxData ticklower tickupper and fee
+     * @return outputValueA liquidity minted
+     * @return outputValueB not used
+     * @return isAsync set to true always
+     */
+
+    function convert(
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata _inputAssetB,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata _outputAssetB,
+        uint256 _inputValue,
+        uint256 _interactionNonce,
+        uint64 _auxData,
+        address
+    )
+        external
+        payable
+        override(IDefiBridge)
+        returns (
+            uint256 outputValueA,
+            uint256 outputValueB,
+            bool isAsync
+        )
+    {
+        if (!(msg.sender == address(ROLLUP_PROCESSOR) || msg.sender == address(this))) revert InvalidCaller();
```

Comment:
Why is it allowed for the msg.sender to be address(this) here? This just seems like a pointless introduction of potential reentrancy vulnerability